docs/shallow_water_equation.md

```diff
@@ -0,0 +1,194 @@
+\int_\Omega \frac{\partial\mathbf{G}}{\partial x}  d\Omega +  &=&
+\int_\Omega \mathbf{S} d\Omega \nonumber\\
+\frac{\partial}{\partial t} \int_\Omega \mathbf{U} d\Omega + 
+\oint_{d\Omega} \left( \mathbf{E}dy  - \mathbf{G} dx \right) ds &=&
+\int_\Omega \mathbf{S} d\Omega \nonumber\\
+ \frac{\partial}{\partial t} \int_\Omega \mathbf{U} d\Omega + 
+\int_{d\Omega} \left( \mathbf{F} . \mathbf{n} \right) ds +  &=&
+\int_\Omega \mathbf{S} d\Omega
+\end{eqnarray}
+
+where
+
+$\mathbf{F}$ is the flux vector and
+$\mathbf{n}$ is the outward pointing unit vector to the boundary $\partial\Omega$. 
+The flux normal flux across the face, $\mathbf{F.n}$, is given by
+
+\begin{align}
+  \mathbf{F.n}
+  =
+  \begin{bmatrix}
+  hu_\perp  \\[.5em]
+  huu_\perp + \frac{1}{2}gh^2 \cos\phi \\[.5em]
+  hvu_\perp + \frac{1}{2}gh^2 \sin\phi 
+  \end{bmatrix}
+\end{align}
+
+where $u_\perp$ is the velocity perpendicular to the face given by $u \cos\phi + v \sin\phi$ and
+$\phi$ is the angle between the face normal and the x axis.
+
+
+### Interface flux
+The interface flux, $\mathbf{F}_\perp$, at the face shared by cell $i$ and $j$ are evaluated using Roe's method as 
+
+\begin{equation}
+\mathbf{F.n} \approx \mathbf{F}_\perp^{i,j} =
+\frac{1}{2} \left( \mathbf{F}_\perp^{i} + \mathbf{F}_\perp^{j} - \mathbf{\hat{R}} |\mathbf{\hat{\Lambda}| \mathbf{\Delta}\hat{V}} \right)
+\end{equation}
+
+where
+
+\begin{align}
+  \mathbf{R}
+  =
+  \begin{bmatrix}
+  1 & 0 & 1  \\[.5em]
+  \hat{u} - \hat{c}\cos\phi & -\sin\phi & \hat{u} + \hat{c}\cos\phi \\[.5em]
+  \hat{v} - \hat{c}\sin\phi & \cos\phi & \hat{v} + \hat{c}\sin\phi
+  \end{bmatrix}
+\end{align}
+
+\begin{align}
+  \mathbf{\Delta\hat{V}}
+  =
+  \begin{bmatrix}
+  \frac{1}{2} \left( \Delta h - \frac{\hat{h}\Delta u_\perp}{\hat{a}} \right) \\[.5em]
+  \hat{h}u_\parallel \\[.5em]
+  \frac{1}{2} \left( \Delta h + \frac{\hat{h}\Delta u_\perp}{\hat{a}} \right)
+  \end{bmatrix}
+\end{align}
+
+\begin{eqnarray}
+  \hat{h} & = & \sqrt{h_i h_j} \\
+  \hat{u} & = & \frac{ \sqrt{h_i} u_i + \sqrt{h_j} u_j}{ \sqrt{h_i} + \sqrt{h_j}} \\
+  \hat{v} & = & \frac{ \sqrt{h_i} v_i + \sqrt{h_j} v_j}{ \sqrt{h_i} + \sqrt{h_j}} \\
+  \hat{a} & = & \sqrt{\frac{g}{2} \left( h_i + h_j \right)}
+\end{eqnarray}
+
+\begin{align}
+  |\mathbf{\hat{\Lambda}}|
+  =
+  \begin{bmatrix}
+  | \hat{u}_\perp - \hat{a} |^* & 0 & 0  \\[.5em]
+  0                                     & |\hat{u}_\perp| & 0 \\[.5em]  
+  0                                     &                         & | \hat{u}_\perp + \hat{a} |^* 
+  \end{bmatrix}
+\end{align}
+
+The asterisks denote that the eigenvalues 
+$\hat{\lambda}_1 (= \hat{u}_\perp - \hat{a} )$ and
+$\hat{\lambda}_3 (= \hat{u}_\perp + \hat{a})$ 
+are adjusted since Roe's method does not provide correct flux for critical flow.
+
+\begin{eqnarray}
+  |\hat{\lambda}|_1 &=& \frac{\hat{\lambda}^2_1}{\Delta \lambda} + \frac{\Delta \lambda}{4} \mbox{\hspace{0.5cm}if $-\Delta \lambda/2 < \hat{\lambda}_1 < \Delta \lambda/2$} \\
```

Comment:
\hspace{0.5cm} is not interpreted.

Reply:
See [this](https://docs.mathjax.org/en/latest/input/tex/macros/index.html) for supported LaTeX commands.